lib/providers/userscripts_provider.dart

```diff
@@ -111,23 +113,41 @@ class UserScriptsProvider extends ChangeNotifier {
     if (!_userScriptsEnabled) {
       return const <String>[];
     } else {
-      return _userScriptList.where((s) => !s.shouldInject(url)).map((s) => s.name).toList();
+      return _userScriptList
+          .where((s) => !s.shouldInject(url))
+          .map((s) => s.name)
+          .toList();
     }
   }
 
-  String adaptSource(String source, String apiKey) {
+  String adaptSource(String source, String apiKey, List<String>? grants) {
     final String withApiKey = source.replaceAll("###PDA-APIKEY###", apiKey);
-    String anonFunction = "(function() {$withApiKey}());";
+    String grantInjections =
+        "var GM = {};"; // should this exist if we have no grants?
+    if (grants?.contains("GM.setValue") ?? false) {
+      grantInjections +=
+          "GM.setValue = function(key, value) {return localStorage.setItem(key, value);};";
+    }
+    if (grants?.contains("GM.getValue") ?? false) {
+      grantInjections +=
+          "GM.getValue = function(key, defaultValue) {return localStorage.getItem(key) ?? defaultValue;};";
```

Comment:
This won't be scalable -- likely what I'll do is have a .js file for each grant and load them all into a map. This would just iterate through that map and if the grant is included inject the right string, making this bit pretty generic. But for the moment, this is where we are.